assets/js/modules/ads/components/common/PAXEmbeddedApp.js

```diff
@@ -149,6 +162,26 @@ export default function PAXEmbeddedApp( {
 		launchPAXApp,
 	] );
 
+	/* eslint-disable react-hooks/exhaustive-deps */
+	// Rule disabled for compiled `paxDateRange` properties, instead of using the main object
+	// to ensure effect runs every time dates are changed.
+	useEffect( () => {
+		if ( displayMode === 'reporting' && paxAppRef?.current ) {
+			const { startDate, endDate } = paxDateRange;
+
+			paxAppRef.current.getServices().adsDateRangeService.update( {
+				startDate: formatPaxDate( startDate ),
+				endDate: formatPaxDate( endDate ),
+			} );
+		}
+	}, [
+		paxAppRef?.current,
```

Comment:
Using a `ref` as a dependency of a `useEffect` is an anti-pattern, because mutating the `ref` doesn't cause the effect run again.

Maybe we should make this `paxAppRef` a state value instead? It will trigger the component to render again, but I don't think that's actually an issue 🤔

Reply:
I've changed this to run the date update code after the app is initialised.

It was possible to re-render when the dates changes if they were destructured, as @benbowler mentioned.

Reply:
> because mutating the ref doesn't cause the effect run again

I thought so, but actually it did re-trigger the hook once `ref.current` was set to the pax Instance, which is not re-rendered otherwise. As it changes value from `undefined` to an app instance, it is causing it to run again